### DISCUSSION TITLE
Will iperf3 support poll or epoll in the future?

### DISCUSSION THREAD

**Discussion by endinferno** [esnet/iperf#1253] [Ideas]
Recently I tried to rewrite 'select' in iperf3 to 'poll' or 'epoll'.
Will iperf3 officially have this plan?

**Comment by bmah888** [comment]
What problem are you trying to solve by switching from select() to poll()? I understand that there are a number of ways that poll() is "better" but I'm trying to see which, if any, are applicable to iperf3 and whether it's worth the effort to do this change. Thanks!

**Reply by endinferno** [reply]
Because I want to port iperf3 to an operating system which have no select, only have poll and epoll.

**Comment by kastiljo** [comment]
Also with select we are bounded to FD_SETSIZE value limitation (1024) which means there is not possible to have more than 500 parallel flows even if we change iperf's internal max value.

**Comment by swlars** [comment]
We have discussed this, but have not looked into it yet. We're going to be looking into how we use select, since there have been a few recent changes to it.

**Reply by kastiljo** [reply]
Great, I hope there will be possibility to have more parallel flows in the future. Thank you for great work.
